### PR TITLE
Collection replace and replaceRecursively functions

### PR DESCRIPTION
**Pull request by salarmehr**

### CODE REVIEW

**Comment by lk77** [comment]
I think replacing in collections is not needed by the framework, 
a PR was declined on this functionality recently if i remember correctly
You should use map function instead.

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -1659,4 +1659,28 @@ public function __get($key)
 
         return new HigherOrderCollectionProxy($this, $key);
     }
+
+    /**
+     * Replaces elements from passed arrays into the collection
+     *
+     * @param  mixed $items
+     * @return static
+     * @see http://php.net/manual/en/function.array-replace-recursive.php
```
Comment (lk77) [inline src/Illuminate/Support/Collection.php]: it's http://php.net/manual/en/function.array-replace.php

**Comment by taylorotwell** [comment]
No plans to add this.

**Comment by salarmehr** [comment]
@taylorotwell 
Thanks for you comment. Can you please give a hint that what sort of things are planed? In the collection there are a number of  non-essential methods (e.g. `median()`) while the "replacing" items is such important that php has two build-in function for that. (or is there any downside for including `replace`)

BTW, the last year that I pushed this for the first time it seems the reaction to that was positive #15157

**Pull request closed without merging**